polars/polars-core/src/chunked_array/ops/cum_agg.rs

```diff
@@ -36,37 +36,29 @@ where
     }
 }
 
-fn det_sum<T>(state: &mut Option<T>, v: Option<T>) -> Option<Option<T>>
+fn det_sum<T>(state: &mut T, v: Option<T>) -> Option<Option<T>>
 where
     T: Copy + PartialOrd + AddAssign + Add<Output = T>,
 {
-    match (*state, v) {
-        (Some(state_inner), Some(v)) => {
-            *state = Some(state_inner + v);
-            Some(*state)
-        }
-        (None, Some(v)) => {
-            *state = Some(v);
-            Some(*state)
+    match v {
+        Some(v) => {
+            *state = *state + v;
+            Some(Some(*state))
         }
-        (_, None) => Some(None),
+        None => Some(None),
     }
 }
 
-fn det_prod<T>(state: &mut Option<T>, v: Option<T>) -> Option<Option<T>>
+fn det_prod<T>(state: &mut T, v: Option<T>) -> Option<Option<T>>
```

Comment:
I think now should return `Option<T>` as the identity isn't `None` anymore, but `0` or `1`.

Reply:
First attempt at this doesn't play nicely with the `collect_trusted` iter.

```sh
error[E0271]: expected `Box<dyn PolarsIterator<Item = Option<<T as PolarsNumericType>::Native>>>` to be an iterator that yields `<T as PolarsNumericType>::Native`, but it yields `Option<<T as PolarsNumericType>::Native>`
  --> polars/polars-core/src/chunked_array/ops/cum_agg.rs:94:59
   |
94 |             false => self.into_iter().scan(init, det_sum).collect_trusted(),
   |                                                           ^^^^^^^^^^^^^^^ expected associated type, found `Option<<... as PolarsNumericType>::Native>`
   |
   = note: expected associated type `<T as datatypes::PolarsNumericType>::Native`
                         found enum `std::option::Option<<T as datatypes::PolarsNumericType>::Native>`
   = help: consider constraining the associated type `<T as datatypes::PolarsNumericType>::Native` to `std::option::Option<<T as datatypes::PolarsNumericType>::Native>` or calling a method that returns `<T as datatypes::PolarsNumericType>::Native`
   = note: for more information, visit https://doc.rust-lang.org/book/ch19-03-advanced-traits.html
   = note: required for `Scan<Box<dyn PolarsIterator<Item = Option<<T as PolarsNumericType>::Native>>>, ..., ...>` to implement `polars_arrow::trusted_len::TrustedLen`
   = note: the full type name has been written to '/home/willayd/clones/polars/target/debug/deps/polars_core-f27f8b388d927e0a.long-type-3305089848937386963.txt'
```

Need to dig a bit further into how that is set up